src/System.Data.SqlClient/src/System/Data/Common/DbConnectionOptions.cs

```diff
@@ -75,7 +75,7 @@ internal bool ConvertValueToIntegratedSecurityInternal(string stringValue)
         public int ConvertValueToInt32(string keyName, int defaultValue)
         {
             string value;
-            return _parsetable.TryGetValue(keyName, out value) ?
+            return _parsetable.TryGetValue(keyName, out value) && !string.IsNullOrWhiteSpace(value) ?
```

Comment:
It looks like the reference source is only checking for null, but this is checking for null and whitespace.  Why the extra whitespace check?